lib/src/components/NxTooltip/__tests__/NxTooltip.test.tsx

```diff
@@ -315,4 +315,147 @@ describe('NxTooltip', function() {
+      });
+
+      it('does not override the accessible name if already set by aria-label or aria-labelledby', async function() {
+        render(<span id="label">baz</span>);
+
+        const user = userEvent.setup(),
+            childrenWithLabelName = <button aria-label="bar">Foo</button>,
+            childrenWithLabelledByName = <button aria-labelledby="label">Foo</button>,
+            viewWithLabelName = quickRender({ children: childrenWithLabelName }),
+            viewWithLabelledByName = quickRender({ children: childrenWithLabelledByName }),
+            buttonWithLabelName = viewWithLabelName.getByRole('button'),
+            buttonWithLabelledByName = viewWithLabelledByName.getByRole('button');
+
+        await user.hover(buttonWithLabelName);
+        await runTimers();
+
+        expect(buttonWithLabelName).toHaveAccessibleName('bar');
+        expect(buttonWithLabelName).not.toHaveAttribute('title');
+
+        await user.hover(buttonWithLabelledByName);
+        await runTimers();
+
+        expect(buttonWithLabelledByName).toHaveAccessibleName('baz');
+        expect(buttonWithLabelledByName).not.toHaveAttribute('title');
+      });
+
+      it('does not override the accessible description and does not set the title', async function() {
+        const user = userEvent.setup(),
+            childrenWithOwnDescription = <button aria-describedby="descriptionId">Foo</button>,
+            childrenWithNoDescription = <button>Foo</button>,
+            viewWithOwnDescription = quickRender({ children: childrenWithOwnDescription }),
+            viewWithNoDescription = quickRender({ children: childrenWithNoDescription }),
+            buttonWithOwnDescription = viewWithOwnDescription.getByRole('button'),
+            buttonWithNoDescription = viewWithNoDescription.getByRole('button');
+
+        expect(buttonWithOwnDescription).toHaveAccessibleDescription('Description Text');
+        expect(buttonWithOwnDescription).not.toHaveAttribute('title');
+        expect(buttonWithNoDescription).not.toHaveAccessibleDescription();
+        expect(buttonWithNoDescription).not.toHaveAttribute('title');
+
+        await user.hover(buttonWithOwnDescription);
+        await runTimers();
+
+        expect(buttonWithOwnDescription).toHaveAccessibleDescription('Description Text');
+        expect(buttonWithOwnDescription).not.toHaveAttribute('title');
+
+        await user.hover(buttonWithNoDescription);
+        await runTimers();
+
+        expect(buttonWithNoDescription).not.toHaveAccessibleDescription();
+        expect(buttonWithNoDescription).not.toHaveAttribute('title');
+      });
+    });
+
+    describe('when title is JSX', function() {
+      const title = <span data-testid="tooltip-content">tiptip</span>,
+          quickRender = rtlRender(NxTooltip, { ...minimalProps, isName: true, title });
+
+      it('sets the title as the accessible name of the child only when hovered', async function() {
```

Comment:
why does this differ from when the title is a string?

Reply:
> why does this differ from when the title is a string?

Admittedly, this test is more of a recording of what MUI happens to do rather than an assertion that this is what we _want_ it to do.  But essentially when you use a JSX tooltip, MUI doesn't make an attempt to work out the text content of that JSX and so can only set the accessible name via `aria-labelledby` once the tooltip is actually rendered.  I imagine they _could_ render into the DOM all the time, and just set it to display: none when not active, in order to set the accessible name when not hovered, but it doesn't seem they've elected to do that.